src/lib.rs

```diff
@@ -201,10 +201,6 @@ impl<T: Clone + Sync> Seat<T> {
                 .expect("seat that should be occupied was empty")
         };
 
-        // let writer know that we no longer need this item.
-        // state is no longer safe to access.
-        #[allow(clippy::drop_ref)]
-        drop(state);
```

Comment:
I kind of like having these explicit drops there to make it clear that you shouldn't be trying to access the value any more here. In some sense, it is my little manual way to highlight the problem that miri is pointing out _with_ the `drop` :sweat_smile: Could we maybe move it into the else clause? I think that should make miri happy?

Reply:
done in https://github.com/jonhoo/bus/pull/29/commits/95aa5bf8b160ee245bcfd3d61c43cbda17663a72